drivers/sensor/default_rtio_sensor.c

```diff
@@ -202,17 +205,17 @@ static void sensor_submit_fallback(const struct device *dev, struct rtio_iodev_s
 				continue;
 			}
 
-			/* Convert the value to micro-units */
-			int64_t value_u = sensor_value_to_micro(&value[sample]);
-
 			/* Convert to q31 using the shift */
 			q[sample_idx + sample] =
-				((value_u * ((INT64_C(1) << 31) - 1)) / 1000000) >> header->shift;
+				((int64_t)value[sample].val1 << float_bits) |
```

Comment:
a bit unfortunate to need 64bit operations to convert to q31, but this being a fallback path seems ok

Reply:
Are we sure we need to? If the shift is proper the integer part of the number should not be going past the 32-bit word limit (and for negative shifts the integer portion should just be zero)

Reply:
So there are a few ways we can do this. Lets take the number 0.75 as an example. Really what we want is:
```
Q = (750000 / 1000000) * BIT(num_float_bits)
```

Clearly that doesn't work since we'd just get 0. The current solution swaps the order but for that to work we have to upcast to 64 bits, otherwise `750000 * BIT(num_float_bits)` would overflow for any `num_float_bits` 12 or greater.

We could get away without upcasting. In theory `val2` should be in the range of -999,999 to 999,999. This means that if `num_float_bits` is 15 for example then each bit is 30.517578125 micro-units, so in theory we could do `val2 / (1000000 / BIT(num_float_bits))` but we'd lose a fair amount of cumulative accuracy there. We could scale things a bit better since we know the bounds of `val2`, but still, it's not as good. This also means that if `num_float_bits` is greater than 20, this math doesn't really work because `1000000 / BIT(20)` is 0. This means that we have to start introducing `if` statements which would degrade performance much more than a 64 bit operation.